qualang_tools/config/waveform_tools.py

```diff
@@ -26,46 +26,24 @@ def drag_gaussian_pulse_waveforms(
     :return: Returns a tuple of two lists. The first list is the 'I' waveform (real part) and the second is the
         'Q' waveform (imaginary part)
     """
-    delta = kwargs.get("delta", None)
-    if delta is not None:
-        print("'delta' has been replaced by 'anharmonicity' and will be deprecated in the future. ")
-        if alpha != 0 and delta == 0:
-            raise Exception("Cannot create a DRAG pulse with `anharmonicity=0`")
-        t = np.arange(length, step=1e9 / sampling_rate)  # An array of size pulse length in ns
-        center = (length - 1e9 / sampling_rate) / 2
-        gauss_wave = amplitude * np.exp(-((t - center) ** 2) / (2 * sigma**2))  # The gaussian function
-        gauss_der_wave = (
-            amplitude * (-2 * 1e9 * (t - center) / (2 * sigma**2)) * np.exp(-((t - center) ** 2) / (2 * sigma**2))
-        )  # The derivative of gaussian
-        if subtracted:
-            gauss_wave = gauss_wave - gauss_wave[-1]  # subtracted gaussian
-        z = gauss_wave + 1j * 0
-        if alpha != 0:
-            # The complex DRAG envelope:
-            z += 1j * gauss_der_wave * (alpha / (delta - 2 * np.pi * detuning))
-            # The complex detuned DRAG envelope:
-            z *= np.exp(1j * 2 * np.pi * detuning * t * 1e-9)
-        I_wf = z.real.tolist()  # The `I` component is the real part of the waveform
-        Q_wf = z.imag.tolist()  # The `Q` component is the imaginary part of the waveform
-    else:
-        if alpha != 0 and anharmonicity == 0:
-            raise Exception("Cannot create a DRAG pulse with `anharmonicity=0`")
-        t = np.arange(length, step=1e9 / sampling_rate)  # An array of size pulse length in ns
-        center = (length - 1e9 / sampling_rate) / 2
-        gauss_wave = amplitude * np.exp(-((t - center) ** 2) / (2 * sigma**2))  # The gaussian function
-        gauss_der_wave = (
-            amplitude * (-2 * 1e9 * (t - center) / (2 * sigma**2)) * np.exp(-((t - center) ** 2) / (2 * sigma**2))
-        )  # The derivative of gaussian
-        if subtracted:
-            gauss_wave = gauss_wave - gauss_wave[-1]  # subtracted gaussian
-        z = gauss_wave + 1j * 0
-        if alpha != 0:
-            # The complex DRAG envelope:
-            z += 1j * gauss_der_wave * (alpha / (2 * np.pi * anharmonicity - 2 * np.pi * detuning))
-            # The complex detuned DRAG envelope:
-            z *= np.exp(1j * 2 * np.pi * detuning * t * 1e-9)
-        I_wf = z.real.tolist()  # The `I` component is the real part of the waveform
-        Q_wf = z.imag.tolist()  # The `Q` component is the imaginary part of the waveform
+    if alpha != 0 and anharmonicity == 0:
+        raise Exception("Cannot create a DRAG pulse with `anharmonicity=0`")
```

Comment:
@TheoLaudatQM But you see raise an error in this case, what am I'm missing?

Reply:
This is if anharmonicity is 0, not detuning